src/overlays/actors/ovl_En_Geg/z_en_geg.c

```diff
@@ -97,87 +124,955 @@ static DamageTable D_80BB3ED4 = {
+        this->actionFunc = func_80BB27D4;
+        this->actor.flags &= ~0x10000;
+    } else if (this->actor.xzDistToPlayer < 150.0f) {
+        this->actor.flags |= 0x10000;
+        func_800B8614(&this->actor, globalCtx, 150.0f);
+    }
+}
+
+void func_80BB31B8(EnGeg* this, GlobalContext* globalCtx) {
+    s32 getItemId = GI_MASK_DON_GERO;
+
+    if (INV_CONTENT(ITEM_MASK_DON_GERO) == ITEM_MASK_DON_GERO) {
+        if (Player_GetMask(globalCtx) == PLAYER_MASK_DON_GERO) {
+            this->unk_496 = 0xD8B;
+            getItemId = GI_RUPEE_PURPLE;
+        } else {
+            this->unk_496 = 0xD73;
+            getItemId = GI_RUPEE_PURPLE;
+        }
+    } else {
+        this->unk_496 = 0xD70;
+    }
+
+    if (Actor_HasParent(&this->actor, globalCtx)) {
+        this->actor.parent = NULL;
+        gSaveContext.weekEventReg[61] |= 1;
+        if (getItemId == GI_MASK_DON_GERO) {
+            this->unk_230 |= 0x40;
+        }
+        this->actionFunc = func_80BB32AC;
+    } else {
+        func_800B8A1C(&this->actor, globalCtx, getItemId, 300.0f, 300.0f);
+    }
+}
+
+void func_80BB32AC(EnGeg* this, GlobalContext* globalCtx) {
+    if (func_800B84D0(&this->actor, globalCtx)) {
+        func_801518B0(globalCtx, this->unk_496, &this->actor);
+        this->actionFunc = func_80BB27D4;
+    } else {
+        func_800B85E0(&this->actor, globalCtx, 400.0f, -1);
+    }
+}
+
+void func_80BB3318(EnGeg* this, GlobalContext* globalCtx) {
+    static Vec3f D_80BB4044[] = {
+        { -550.0f, 8.0f, 550.0f },
+        { 220.0f, 43.0f, 525.0f },
+    };
+    s16 sp46;
+    f32 sp40;
+
+    if (this->unk_4D8 < 2) {
+        sp46 = Math_Vec3f_Yaw(&this->actor.world.pos, &D_80BB4044[this->unk_4D8]);
+        sp40 = Math_Vec3f_DistXZ(&this->actor.world.pos, &D_80BB4044[this->unk_4D8]);
+        Math_SmoothStepToS(&this->actor.world.rot.y, sp46, 4, 1000, 1);
+        if (sp40 < 20.0f) {
+            this->unk_4D8++;
+        }
+        this->actor.shape.rot.y = this->actor.world.rot.y;
+        func_800AE930(&globalCtx->colCtx, Effect_GetParams(this->unk_4DC), &this->actor.world.pos, 18.0f,
+                      this->actor.shape.rot.y, this->actor.floorPoly, this->actor.floorBgId);
+    }
+
+    if (ActorCutscene_GetCurrentIndex() != this->unk_49C[7]) {
+        func_800AEF44(Effect_GetParams(this->unk_4DC));
+        Actor_MarkForDeath(&this->actor);
+    } else {
+        Math_ApproachF(&this->actor.speedXZ, 10.0f, 0.2f, 1.0f);
+        Actor_SetVelocityAndMoveYRotationAndGravity(&this->actor);
+    }
+
+    func_800B9010(&this->actor, NA_SE_EN_GOLON_SIRLOIN_ROLL - SFX_FLAG);
+}
+
+void func_80BB347C(EnGeg* this, GlobalContext* globalCtx) {
+    if (DECR(this->unk_244) == 0) {
+        this->unk_244 = 65;
+        this->actionFunc = func_80BB221C;
+    }
+}
+
+void EnGeg_Init(Actor* thisx, GlobalContext* globalCtx) {
+    s32 pad;
+    EnGeg* this = THIS;
+    s32 pad2;
+    s32 sp34[] = {
+        0x3E,
+        0xF64,
+    };
```

Comment:
Could you remove the trailing comma and reformat?

Reply:
Agreed, but it doesn't need to hold up the merge.